api.visual/apichanges.xml

```diff
@@ -29,7 +29,20 @@
     </apidefs>
 
     <changes>
-
+        <change>
+            <api name="general"/>
+            <summary>States for non-registered objects</summary>
+            <version major="2" minor="49"/>
+            <date day="27" month="9" year="2017"/>
+            <author login="sdedic"/>
+            <compatibility addition="yes"/>
```

Comment:
Is it really compatible? From the wording you use I assume there are cases where the observable behavior may suddenly change. As such I'd suggest to consider an *optional buy in* - e.g. add a constructor/method parameter, setter, etc. that one could use to turn this behavior on.

Reply:
Hm. Some cases, which were not permitted previously, i.e. setSelectedObjects threw NPE on iterating object2widgets.get (object), are not OK. Behaviour for previously supported cases did not change AFAIK, so existing code should not be affected at all.

Reply:
OK, I think that existing **working** code will not be affected by the change. In the original state, an attempt to select or highlight an object without widgets would result in NPE; so working code must have these parts fixed.
I really do not expect existing code to be broken by these changes.